File: 哈希表/187. 重复的DNA序列/重复的DNA序列.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;
// 时间复杂度O(n)，空间复杂度O(n)

class Solution
{
public:
    vector<string> findRepeatedDnaSequences(string s)
    {
        unordered_map<string, int> repetitivestring;
        vector<string> arr;
        for (int i = 0; i + 10 <= s.size(); i++)
        {
            string temp = s.substr(i, 10);
            repetitivestring[temp]++;
            if (repetitivestring[temp] == 2)
            {
                arr.push_back(temp);
            }
        }
        return arr;
    }
};

File: 哈希表/187. 重复的DNA序列/重复的DNA序列.js
/**
 * @param {string} s
 * @return {string[]}
 */
var findRepeatedDnaSequences = function (s) {
  let repetitivestring = new Map();
  let arr = [];
  let temp;
  for (let i = 0; i + 10 <= s.length; i++) {
    temp = s.slice(i, i + 10);
    repetitivestring.set(temp, (repetitivestring.get(temp) || 0) + 1);
    if (repetitivestring.get(temp) === 2) {
      arr.push(temp);
    }
  }
  return arr;
};
